examples/fizzbuzz/main.c

```diff
@@ -0,0 +1,189 @@
+#include <qmlbind.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/*!
+ * \file main.c
+ * \brief This example shows how to wrap fizzbuzz into a metaclass with `libqmlbind`.
+ */
+
+#define ENOUGH 10 // FIXME: use correct value
+
+
+typedef struct fizzbuzz
+{
+    int input;
+    char *result;
+    qmlbind_signal_emitter *signalEmitter;
+} fizzbuzz_t;
+
+
+void on_changed(fizzbuzz_t *fb, const qmlbind_value *input)
+{
+    int i = (int) qmlbind_value_get_number(input);
+    if (i % 15 == 0) {
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "inputWasFizzBuzz",
+                                    0, NULL);
+        strcpy(fb->result, "FizzBuzz");
+    } else if (i % 3 == 0) {
+        strcpy(fb->result, "Fizz");
+    } else if (i % 5 == 0) {
+        strcpy(fb->result, "Buzz");
+    } else {
+        sprintf(fb->result, "%d", i);
+    }
+
+    qmlbind_value params[1] = { qmlbind_value_new_string(strlen(fb->result), fb->result) };
+    qmlbind_signal_emitter_emit(fb->signalEmitter, "resultChanged", 1, params);
+}
+
+
+qmlbind_client_object *
+fizzbuzz_new_object_cb(qmlbind_client_class *classRef,
+                       qmlbind_signal_emitter *signalEmitter)
+{
+    fizzbuzz_t *fb = malloc(sizeof(fizzbuzz_t));
+    fb->input = 0;
+    fb->result = malloc(sizeof(char) * ENOUGH);
+    strcpy(fb->result, "");
+    fb->signalEmitter = signalEmitter;
+
+    return (qmlbind_client_object *) fb;
+}
+
+
+void fizzbuzz_delete_object_cb(qmlbind_client_object *obj)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+    free(fb->result);
+    qmlbind_signal_emitter_release(fb->signalEmitter);
+    free(fb);
+}
+
+
+qmlbind_value *fizzbuzz_call_method_cb(qmlbind_engine *engine,
+                                       qmlbind_client_object *obj,
+                                       const char *name,
+                                       int argc,
+                                       const qmlbind_value *const *argv)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "onInputChanged") == 0 && argc == 1) {
```

Comment:
Even if it could be discarded... How do I connect `onInputChanged` so it reacts to `inputChanged` signal? Calling it directly seems like a workaround to me...